### PR TITLE
clean up a few tests

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Remove a few redundant test cases and suite. Changes a few tests to use "persistent" indexes instead of "hash"/"skiplist".
This PR only contains changes to test code.
There is no plan to backport these changes.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -
  - [ ] Backport for 3.7: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: tests/js/server/aql/aql-queries-optimizer-in-noncluster.js [hunk 1]
```diff
@@ -28,49 +28,36 @@
...
+const db = internal.db;
 
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief test suite
 ////////////////////////////////////////////////////////////////////////////////
 
 function ahuacatlQueryOptimizerInTestSuite () {
-  var c = null;
-  var cn = "UnitTestsAhuacatlOptimizerIn";
+  const cn = "UnitTestsAhuacatlOptimizerIn";
```
Comment (dothebart) [inline tests/js/server/aql/aql-queries-optimizer-in-noncluster.js:42]: Ahuacatl was removed in other parts of this diff?

File: tests/js/server/aql/aql-queries-optimizer-limit-noncluster.js [hunk 2]
```diff
@@ -28,44 +26,38 @@
...
 
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief test suite
 ////////////////////////////////////////////////////////////////////////////////
 
 function ahuacatlQueryOptimizerLimitTestSuite () {
-  var collection = null;
-  var docCount = 100;
-  var cn = "UnitTestsAhuacatlOptimizerLimit";
+  const cn = "UnitTestsAhuacatlOptimizerLimit";
```
Comment (dothebart) [inline tests/js/server/aql/aql-queries-optimizer-limit-noncluster.js:40]: Ahuacatl

File: tests/js/server/aql/aql-queries-optimizer-sort.js [hunk 3]
```diff
@@ -28,52 +26,44 @@
...
 
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief test suite
 ////////////////////////////////////////////////////////////////////////////////
 
 function ahuacatlQueryOptimizerSortTestSuite () {
-  var collection = null;
-  var cn = "UnitTestsAhuacatlOptimizerSort";
-  var idx = null;
+  const cn = "UnitTestsAhuacatlOptimizerSort";
```
Comment (dothebart) [inline tests/js/server/aql/aql-queries-optimizer-sort.js:42]: Ahuacatl?

File: tests/js/server/aql/aql-queries-optimizer-sort.js [hunk 4]
```diff
@@ -4,8 +4,6 @@
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief tests for query language, sort optimizations
 ///
```
Comment (dothebart) [inline tests/js/server/aql/aql-queries-optimizer-sort.js:6]: why is this one removed from cluster to non-cluster? or is this just a github diff artifact?
 196  ...aql/aql-queries-optimizer-sort-cluster.js → .../server/aql/aql-queries-optimizer-sort.js

**Review by dothebart — APPROVED** [review]
Please clarify the naming, and the one cluster/noncluster test 
else LGTM.

File: tests/js/server/aql/aql-queries-optimizer-in-noncluster.js (hunk 1 above)
Comment (jsteemann) [reply tests/js/server/aql/aql-queries-optimizer-in-noncluster.js:42]: It's just a name. It doesn't really matter IMO.

File: tests/js/server/aql/aql-queries-optimizer-limit-noncluster.js (hunk 2 above)
Comment (jsteemann) [reply tests/js/server/aql/aql-queries-optimizer-limit-noncluster.js:40]: It's just a name. It doesn't really matter IMO.

File: tests/js/server/aql/aql-queries-optimizer-sort.js (hunk 3 above)
Comment (jsteemann) [reply tests/js/server/aql/aql-queries-optimizer-sort.js:42]: It's just a name. It doesn't really matter IMO.

File: tests/js/server/aql/aql-queries-optimizer-sort.js (hunk 4 above)
Comment (jsteemann) [reply tests/js/server/aql/aql-queries-optimizer-sort.js:6]: We had several tests that were split into cluster and non-cluster.
That was done years ago when single server and cluster had vastly different feature sets.
For most of the tests this split wasn't justified anymore, and we can actually run the same test in both cluster and non-cluster mode.

Comment (dothebart) [reply tests/js/server/aql/aql-queries-optimizer-sort.js:6]: OK.

**Pull request merged**